llvm/lib/Target/AArch64/AArch64ISelDAGToDAG.cpp

```diff
@@ -7379,13 +7379,27 @@ bool AArch64DAGToDAGISel::SelectAddrModeIndexedSVE(SDNode *Root, SDValue N,
   if (N.getOpcode() != ISD::ADD)
     return false;
 
-  SDValue VScale = N.getOperand(1);
-  if (VScale.getOpcode() != ISD::VSCALE)
+  int64_t MulImm = std::numeric_limits<int64_t>::max();
+  if (SDValue VScale = N.getOperand(1); VScale.getOpcode() == ISD::VSCALE)
```

Comment:
I'd prefer the declaration of `VScale` to be moved out above the if statement for two reasons:

1. I personally find it hard to parse with the semi-colon because even outside of 'if' statements we don't normally have multiple statements on a single line.
2. You can reuse `VScale` in the else if case below, i.e. `auto C = dyn_cast<ConstantSDNode>(VScale)`

Reply:
Thank you for the suggestion, done!